Fabric.Authorization.AccessControl/src/app/modules/access-control/custom-group/custom-group.component.ts

```diff
@@ -280,37 +247,28 @@ export class CustomGroupComponent implements OnInit, OnDestroy {
       .getChildGroups(this.groupName, this.identityProvider, this.tenantId);
   }
 
-  getRolesBySecurableItemAndGrain(): Observable<IRole[]> {
-    return this.roleService
-      .getRolesBySecurableItemAndGrain(
-        this.grain,
-        this.securableItem
-      );
-  }
-
-  getGroupRoles(): Observable<IRole[]> {
-     return this.groupService
-      .getGroupRoles(
-        this.groupName,
-        this.grain,
-        this.securableItem,
-        this.identityProvider,
-        this.tenantId
-      );
+  getGroup(): Observable<IGroup> {
+     return this.groupService.getGroup(this.groupName);
   }
 
   getGroupRolesBySecurableItemAndGrain(): Observable<IRole[]> {
-    return observableZip(this.getRolesBySecurableItemAndGrain(), this.getGroupRoles()).pipe(
-      map((result: [IRole[], IRole[]]) => {
-        let allRoles = result[0];
-        this.groupRoles = result[1];
+    return observableZip(this.roleService.getRolesBySecurableItemAndGrain(this.grain, this.securableItem), this.getGroup()).pipe(
```

Comment:
From above this is getting all groups with that groupName, then you filter down based on grain and securableItem.  Wouldnt it be 1 item already?  

But thats probably needed because of how everything works in lists with observables.  if you have something left over then you have permissions, but otherwise reject it.

Reply:
Which method are you referring to? `getGroup` will always return a single group. I'm filter the roles by grain and securableItem.